File: src/main/java/com/kenvix/utils/exception/LifecycleEndException.java
package com.kenvix.utils.exception;

import java.util.concurrent.CancellationException;

public class LifecycleEndException extends CancellationException {
    public LifecycleEndException() {
        super("Lifecycle of the activity or service, etc. is end, so this job is cancelled");
    }

    public LifecycleEndException(String message) {
        super(message);
    }
}

File: src/main/java/com/kenvix/utils/exception/LifecycleEndException.kt
package com.kenvix.utils.exception

import java.util.concurrent.CancellationException

class LifecycleEndException : CancellationException {
    constructor() : super("Lifecycle of the activity or service, etc. is end, so this job is cancelled") {}
    constructor(message: String?) : super(message)
}